src/activity.rs

```diff
@@ -52,7 +56,26 @@ impl ActivityMask {
         variances.push(variance);
       }
     }
-    ActivityMask { variances: variances.into_boxed_slice() }
+
+    // Compute initial segmentation choices for variance AQ
+    // We take the 95th percentile of variances here to get a less skewed sample.
+    // Otherwise segment 0 becomes heavily oversaturated.
```

Comment:
So one of the original problems with the initial patch is selecting segments this way isn't compatible with e.g. selecting segments for Temporal RDO.

I don't quite understand how this patch is trying to resolve that. I see it's creating a list of segments for the variance, and then trying to push around what segments of those are used for temporal RDO. This seems unlikely to produce comparable temporal RDO results when RDO is on. Is that correct?

Reply:
This portion of the code calculates the variance portion of the segmentation decision, but in `segmentation_optimize_aq` is where the final segments are calculated by weighting the variance and the temporal RDO decisions together.

Reply:
Basically, the thing I don't like here is the way we produce the segmentation binning. I would expect the initial segment splits (this is producing 8 of them split equally among the variance range, as I read) to be produced *after* the variances are combined with the temporal RDO decisions.

Reply:
I see what you're saying. I don't think the way I'm doing the binning is "wrong" per se, but your method does sound likely to be better. I'll try to put something together where we merge the variance and RDO results before binning.

Reply:
I guess another issue is us using the max variance of the frame. This is going to vary from frame to frame and end up with our segmentation map changing a lot (if I'm reading this right?). Using percentiles helps, but another option might just be to use a fixed max variance entirely (and therefore, effectively fixed segment map). Some frames with low variance might not use all the segments, but I think that's an ok tradeoff.